bumble/device.py

```diff
@@ -5457,9 +5456,16 @@ def on_connection(
         self_resolvable_address: Optional[hci.Address],
         peer_resolvable_address: Optional[hci.Address],
         role: hci.Role,
```

Comment:
It looks like `role` should be `Optional[hci.Role]`, since for incomplete connections, the role isn't yet known and the event will be emitted with `None` as the `role` parameter.